content/master/concepts/compositions.md

```diff
@@ -0,0 +1,993 @@
+      patches:
+        - type: PatchSet
+          patchSetName: reusable-patch
+    - name: second-resource
+      base:
+      # Removed for Brevity
+      patches:
+        - type: PatchSet
+          patchSetName: reusable-patch
+```
+
+#### Patch with EnvironmentConfigs 
+
+Crossplane uses EnvironmentConfigs as an in-memory data store. Compositions can
+read and write from this data store as part of the patch process. 
+
+{{<hint "important" >}}
+EnvironmentConfigs are an alpha feature. Alpha features aren't enabled by
+default. 
+{{< /hint >}}
+
+ EnvironmentConfigs can predefine data that Compositions can use
+ or a Composite Resource can write data to the EnvironmentConfig for other
+ resources to read. 
+
+<!-- vale off -->
+{{< hint "note" >}}
+<!-- vale on --> 
+Read the [EnvironmentConfigs]({{<ref "./environment-configs" >}}) page for 
+more information on using EnvironmentConfigs.
+{{< /hint >}}
+
+To apply a patch using EnvironmentConfigs, first define which EnvironmentConfig
+to use with 
+{{<hover label="envselect"
+line="6">}}environment.environmentConfigs{{</hover>}}. 
+
+<!-- vale Google.Quotes = NO -->
+<!-- vale gitlab.SentenceLength = NO -->
+<!-- ignore false positive -->
+Use either a
+[reference]({{<ref "./managed-resources#matching-by-name-reference" >}})
+or a [selector]({{<ref "./managed-resources#matching-by-selector" >}}) to
+identify the EnvironmentConfig to use.
+<!-- vale Google.Quotes = YES -->
+
+```yaml {label="envselect",copy-lines="none"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+# Removed for Brevity
+spec:
+  environment:
+    environmentConfigs:
+      - ref:
+          name: example-environment
+  resources:
+  # Removed for Brevity
+```
+
+Inside the {{<hover label="envpatch" line="16">}}patches{{</hover>}} of the 
+resource, use 
+{{<hover label="envpatch" line="17">}}ToEnvironmentFieldPath{{</hover>}} to copy
+data from the resource to the EnvironmentConfig.  
+Use {{<hover label="envpatch" line="20">}}FromEnvironmentFieldPath{{</hover>}}
+to copy data to the resource from the EnvironmentConfig.
+
+```yaml {label="envpatch",copy-lines="none"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+# Removed for Brevity
+spec:
+  environment:
+  # Removed for Brevity
+  resources:
+  # Removed for Brevity
+    - name: vpc
+      base:
+        apiVersion: ec2.aws.upbound.io/v1beta1
+        kind: VPC
+        spec:
+          forProvider:
+            cidrBlock: 172.16.0.0/16
+      patches:
+        - type: ToEnvironmentFieldPath
+          fromFieldPath: status.atProvider.id
+          toFieldPath: vpcId
+        - type: FromEnvironmentFieldPath
+          fromFieldPath: tags
+          toFieldPath: spec.forProvider.tags
+```
+
+The [EnvironmentConfigs]({{<ref "./environment-configs" >}}) page has 
+more information on EnvironmentConfigs options and usage.
+
+### Composite functions
+
+Composite functions (`XFNs`) are containers executing any code you define.
```

Comment:
FWIW I'm not 100% sure we want to brand these as "xfns" - that has been kind of organic. If we do I think we'd go with lowercase `XFn` or `xfn`.

Reply:
Also maybe gloss over the whole "are containers" thing given that we're planning to tweak that for beta.